### PR TITLE
CGE-133 Creating entities for chat implementation

### PR DESCRIPTION
**Pull request by drw85**
In order to implement the chat, entities were created: Chat, Message, Attachment. Also, changes were made to the classes that were affected.

### CODE REVIEW

File: src/main/java/com/hillel/items_exchange/model/Attachment.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+@Entity
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+public class Attachment {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+    @ManyToOne(cascade = CascadeType.ALL)
```
Comment (rpkyrych) [inline src/main/java/com/hillel/items_exchange/model/Attachment.java:19]: Why cascade is here but not into Message?

File: src/main/java/com/hillel/items_exchange/model/Message.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    @JoinColumn(name = "user_id")
+    private User user;
+    private String text;
+    @OneToMany(mappedBy = "message", cascade = CascadeType.ALL, orphanRemoval = true)
+    List<Attachment> attachments;
+
+    @PrePersist
+    private void onPrePersist() {
+        setCreated(LocalDateTime.now());
+        setUpdated(LocalDateTime.now());
```
Comment (rpkyrych) [inline src/main/java/com/hillel/items_exchange/model/Message.java:31]: For setting time during create or update operations we have CustomDateTimeProvider. You could update it for LocalDateTime and tried if it works.

File: src/main/java/com/hillel/items_exchange/model/Message.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        setStatus(Status.NEW);
+    }
+
+    @PreUpdate
+    private void onPreUpdate() {
+        setUpdated(LocalDateTime.now());
```
Comment (rpkyrych) [inline src/main/java/com/hillel/items_exchange/model/Message.java:37]: The same idea about CustomDateTimeProvider

File: src/test/java/com/hillel/items_exchange/controller/AdvertisementControllerIntegrationTest.java [hunk 4]
```diff
@@ -1,7 +1,9 @@
 package com.hillel.items_exchange.controller;
 
 import com.github.database.rider.core.api.dataset.DataSet;
+import com.github.database.rider.core.api.dataset.DataSetFormat;
 import com.github.database.rider.core.api.dataset.ExpectedDataSet;
+import com.github.database.rider.core.api.exporter.ExportDataSet;
```
Comment (rpkyrych) [inline src/test/java/com/hillel/items_exchange/controller/AdvertisementControllerIntegrationTest.java:6]: Excess imports

**Review by Jack11M — APPROVED** [review]

**Review by svyatoslav145 — APPROVED** [review]

File: src/main/java/com/hillel/items_exchange/model/Attachment.java (hunk 1 above)
Comment (drw85) [reply src/main/java/com/hillel/items_exchange/model/Attachment.java:19]: Traces of long attempts to make it work))) Changed. Removed to Message class.

File: src/test/java/com/hillel/items_exchange/controller/AdvertisementControllerIntegrationTest.java (hunk 4 above)
Comment (drw85) [reply src/test/java/com/hillel/items_exchange/controller/AdvertisementControllerIntegrationTest.java:6]: Sorry. The same traces))) Will be removed.

File: src/main/java/com/hillel/items_exchange/model/Message.java (hunk 2 above)
Comment (drw85) [reply src/main/java/com/hillel/items_exchange/model/Message.java:31]: Ok. Will try this.

File: src/main/java/com/hillel/items_exchange/model/Message.java (hunk 3 above)
Comment (drw85) [reply src/main/java/com/hillel/items_exchange/model/Message.java:37]: Yeap

**Review by rpkyrych — APPROVED** [review]
Great job! 🥳

**Pull request merged**